### PR TITLE
feat: remove generic errors

### PR DESCRIPTION
**Pull request by dsherret**
This will help us do stuff like retry resolution reloading package information in the CLI.

### CODE REVIEW

File: src/registry.rs [hunk 1]
```diff
@@ -234,28 +263,22 @@ pub trait NpmRegistryApi: Sync {
...
-  ) -> Result<Arc<NpmPackageInfo>, anyhow::Error> {
-    let maybe_package_info = self.maybe_package_info(name).await?;
-    match maybe_package_info {
-      Some(package_info) => Ok(package_info),
-      None => anyhow::bail!("npm package '{}' does not exist", name),
-    }
-  }
+  ) -> Result<Arc<NpmPackageInfo>, NpmRegistryPackageInfoLoadError>;
 
   async fn package_version_info(
```
Comment (dsherret) [inline src/registry.rs:271]: Actually, I think I should remove this from the trait.

**Review by bartlomieju — APPROVED** [review]
LGTM, great to see moving away from `AnyError`

**Comment by dsherret** [comment]
@bartlomieju sorry, I thought of one more commit after. It's better and allowed me to remove some cli specific error messages https://github.com/denoland/deno_npm/pull/2/commits/cd770c060db554ad4f821a346f07f1f6b6cd80d8

**Review by bartlomieju — APPROVED** [review]
LGTM again

**Pull request merged**